### PR TITLE
feat: v2 teams CRUD

### PR DESCRIPTION
**Pull request by supalarry**
Linear [CAL-4830](https://linear.app/calcom/issue/CAL-4830/platform-feat-teams-endpoints)

### CODE REVIEW

File: packages/platform/types/teams/outputs/team.output.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+export class TeamOutputDto {
+  @IsInt()
+  @Expose()
+  @DocsProperty()
+  readonly id!: number;
+
+  @IsInt()
+  @IsOptional()
+  @Expose()
+  readonly parentId?: number;
```
Comment (ThyMinimalDev) [inline packages/platform/types/teams/outputs/team.output.ts:14]: missing ApiPropertyOptional

File: apps/api/v2/test/fixtures/repository/team.repository.fixture.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class TeamRepositoryFixture {
   }
 
   async delete(teamId: Team["id"]) {
-    return this.prismaWriteClient.team.delete({ where: { id: teamId } });
+    return this.prismaWriteClient.team.deleteMany({ where: { id: teamId } });
```
Comment (ThyMinimalDev) [inline apps/api/v2/test/fixtures/repository/team.repository.fixture.ts:24]: deleteMany ? is the return type more adequate ?

Comment (supalarry) [reply apps/api/v2/test/fixtures/repository/team.repository.fixture.ts:24]: I put deleteMany because it does not error if team does not exist

Comment (supalarry) [reply apps/api/v2/test/fixtures/repository/team.repository.fixture.ts:24]: actually lemme put it back as it was

File: packages/platform/types/teams/outputs/team.output.ts (hunk 1 above)
Comment (supalarry) [reply packages/platform/types/teams/outputs/team.output.ts:15]: Well spotted - fixed in [fix: team parentId docs](https://github.com/calcom/cal.com/pull/18019/commits/42da64495858e4191734dde3036f146a90fb5695)

**Review by ThyMinimalDev — APPROVED** [review]

**Pull request merged**